subsys/bootloader/cmake/sign.cmake

```diff
@@ -4,61 +4,75 @@
 # SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
 #
 
-set(sign_depends kernel_elf ${KEY_FILE_DEPENDS})
 
-set(hash_file firmware.sha256)
-set(signature_file firmware.signature)
-set(SIGNATURE_PUBLIC_KEY_FILE public.pem)
+#TODO: Detect what kernel_elf to depend on and resolve what
+#      image should be hashed and signed by b0.
+
+set(KERNEL_ELF_TO_BE_SIGNED kernel_elf)
+
+set(GENERATED_FILES_PATH ${PROJECT_BINARY_DIR}/nrf/subsys/bootloader/generated)
+# This is needed for make, ninja is able to resolve this but make is not
+add_custom_target(
```

Comment:
I have been told that it is bad practice to do a command inside of a `add_custom_target`, it messes with the dependencies AFAIK, and one should use `add_custom_command` which outputs a file to which there is a `add_custom_target` which depends on that file.